store/src/java/com/zimbra/cs/mailbox/MailSender.java

```diff
@@ -427,14 +444,39 @@ public ItemId sendMimeMessage(OperationContext octxt, Mailbox mbox, MimeMessage
         if (identityId != null) {
             identity = Provisioning.getInstance().get(authuser, Key.IdentityBy.id, identityId);
         }
+        return identity;
+    }
+
+    public ItemId sendMimeMessage(OperationContext octxt, Mailbox mbox, MimeMessage mm, List<Upload> uploads,
+            ItemId origMsgId, String replyType, String identityId, boolean replyToSender, MimeProcessor mimeProc) throws ServiceException {
+        Identity identity = fetchIdentity(octxt, identityId, mbox);
         return sendMimeMessage(octxt, mbox, null, mm, uploads, origMsgId, replyType, identity, replyToSender, mimeProc);
     }
 
+    public ItemId sendMimeMessage(OperationContext octxt, Mailbox mbox, MimeMessage mm, List<Upload> uploads,
+            ItemId origMsgId, String replyType, String identityId, boolean replyToSender, MimeProcessor mimeProc,
+            boolean deliveryReport) throws ServiceException {
+        Identity identity = fetchIdentity(octxt, identityId, mbox);
+        return sendMimeMessage(octxt, mbox, null, mm, uploads, origMsgId, replyType, identity, replyToSender, mimeProc, deliveryReport);
+    }
```

Comment:
refactoring lines 438-461: execute all necessary process in the 2nd `sendMimeMessage` defined at line 456. The 1st (at line 450) calls the 2nd. `fetchIdentity` can be removed.

Reply:
1st doesn’t call the 2nd rather calls the method at line `539` so `fetchIdentity` shouldn’t be removed. Also, as the code is open-source, we need to maintain the old method signatures.

Reply:
Done